### PR TITLE
Changes to get LU_OP working

### PR DESCRIPTION
**Pull request by stevenireeves**
@micmelesse checkout these changes to your branch. With this it compiles and runs the tests successfully.

### CODE REVIEW

File: tensorflow/core/kernels/linalg/BUILD [hunk 1]
```diff
@@ -268,9 +268,12 @@ tf_kernel_library(
 tf_kernel_library(
     name = "lu_op",
     prefix = "lu_op",
-    deps = if_cuda([
-        "//tensorflow/core/util:cuda_solvers",
+    deps = if_cuda_or_rocm([
```
Comment (stevenireeves) [inline tensorflow/core/kernels/linalg/BUILD:271]: These changes to get it compiling.

File: tensorflow/core/kernels/linalg/lu_op_gpu.cu.cc [hunk 2]
```diff
@@ -182,9 +182,16 @@ class LuOpGpu : public AsyncOpKernel {
       auto packed_triangular_factors_ptrs = solver->GetScratchSpace<uint8>(
           sizeof(Scalar*) * batch_size, "packed_triangular_factors_ptrs",
           /* on_host */ true);
+#if GOOGLE_CUDA
```
Comment (stevenireeves) [inline tensorflow/core/kernels/linalg/lu_op_gpu.cu.cc:185]: These changes to squash unidentified function GetrsBatched.

File: tensorflow/core/kernels/linalg/BUILD [hunk 3]
```diff
@@ -268,9 +268,12 @@ tf_kernel_library(
...
         "//tensorflow/core/kernels:transpose_functor",
+    ]) + if_cuda([
+        "//tensorflow/core/util:cuda_solvers",
+    ]) + if_rocm([
+        "//tensorflow/core/util:rocm_solvers",
```
Comment (stevenireeves) [inline tensorflow/core/kernels/linalg/BUILD:276]: Above changes to get Bazel to recognize deps.

File: tensorflow/core/kernels/linalg/lu_op_gpu.cu.cc [hunk 4]
```diff
@@ -182,9 +182,16 @@ class LuOpGpu : public AsyncOpKernel {
...
       const Scalar** packed_triangular_factors_ptrs_base =
           reinterpret_cast<const Scalar**>(
               packed_triangular_factors_ptrs.mutable_data());
+#else //TENSORFLOW_USE_ROCM
+       Scalar** packed_triangular_factors_ptrs_base =
+          reinterpret_cast< Scalar**>(
+              packed_triangular_factors_ptrs.mutable_data());
+#endif
+
```
Comment (stevenireeves) [inline tensorflow/core/kernels/linalg/lu_op_gpu.cu.cc:194]: Above changes to get Bazel to stop complaining about unidentified function GetrsBatched. To be explicit, cudaSolvers expects this variable to have the const modifier, whereas rocSolver does not expect const.

**Review by micmelesse — APPROVED** [review]
LGTM

**Pull request merged**